### DISCUSSION TITLE
Are OUTER joins possible in view collections?

### DISCUSSION THREAD

**Discussion by dschissler** [pocketbase/pocketbase#5228] [Q&A] [answered]
Are OUTER joins possible in view collections?

**Comment by ganigeorgiev** [comment, marked as answer]
The identifiers parser currently doesn't explicitly targets `FULL JOIN`/`FULL OUTER JOIN` but technically it should be supported as it would match the "outer join" part. If it doesn't work, you can try with wrapping your query in a sub select (ex. `SELECT id, someField FROM (yourQuery)`). With the refactoring ~I'll add~ (_done + ignoring whitespaces between the multipart keywords_) an explicit entry for the `FULL *` and `CROSS` joins to handle all cases.

**Reply by dschissler** [reply]
The subquery solution works.  It just requires listing the fields twice; once in the top-level SELECT, and once in the subquery.

My view is putting out null values now though.  That's not a problem for me, but I'm used to never having null values anywhere due to Pocketbase not allowing them.  That might be worth thinking about in terms of it creating edge cases in the API related event hooks.

**Reply by ganigeorgiev** [reply]
> My view is putting out null values now though. 

This is working correctly and it happens because the resulting fields of a subquery, or any other custom expression, is resolved to a `json` field.
